superset-frontend/src/filters/components/TimeColumn/TimeColumnFilterPlugin.tsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Behavior, DataMask, styled, t } from '@superset-ui/core';
+import React, { useEffect, useState } from 'react';
+import { Select } from 'src/common/components';
+import { PluginFilterTimeColumnProps } from './types';
+import { PluginFilterStylesProps } from '../types';
+
+const Styles = styled.div<PluginFilterStylesProps>`
+  height: ${({ height }) => height};
+  width: ${({ width }) => width};
+`;
+
+const { Option } = Select;
+
+export default function PluginFilterTimeColumn(
+  props: PluginFilterTimeColumnProps,
+) {
+  const { behaviors, data, formData, height, width, setDataMask } = props;
+  const { defaultValue, currentValue, inputRef } = formData;
+
+  const [value, setValue] = useState<string[]>(defaultValue ?? []);
+
+  const handleChange = (value?: string[] | string | null) => {
+    let resultValue: string[];
+    if (Array.isArray(value)) {
+      resultValue = value;
+    } else {
+      resultValue = value ? [value] : [];
+    }
+    setValue(resultValue);
+
+    const dataMask = {
+      extraFormData: {
```

Comment:
may be move it to utils like in SelectFilter, to re use it in other filters if needed

Reply:
This feels so simple I'd almost prefer to keep this here for now (we can look at these collectively once all filters are in)